### PR TITLE
Fix/invalid order after drag and drop

### PR DESCRIPTION
**Pull request by dangmarm**

### CODE REVIEW

File: src/components/helpers/array2d.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  removeRow(index = this.getRowCount() - 1) {
+    this.array2d.splice(index, 1);
+  }
+
+  insertColumn(index = this.getColumnCount()) {
+    for (let i = 0; i < this.array2d.length; i++) {
+      this.array2d[i].splice(index, 0, undefined);
+    }
```
Comment (ncardeli) [inline src/components/helpers/array2d.ts:84]: Just a suggestion: you can use the newer for...of statement in these cases:

```suggestion
    for (const arr of this.array2d) {
      arr.splice(index, 0, undefined);
    }
```

File: src/components/helpers/array2d.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  removeColumn(index = this.getColumnCount() - 1) {
+    for (let i = 0; i < this.array2d.length; i++) {
+      this.array2d[i].splice(index, 1);
+    }
```
Comment (ncardeli) [inline src/components/helpers/array2d.ts:90]: Same with this one 👍

File: src/components/helpers/array2d.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
+export class Array2d<T> {
```
[the remark this answers is not in the document]
Comment (dangmarm) [reply src/components/helpers/array2d.ts:1]: Please review if it is expected before continuing with CssGridLayout

[the remark this answers is not in the document]
Comment (ncardeli) [reply src/components/helpers/array2d.ts:1]: Amazing job! 💪

**Pull request merged**